### ISSUE TITLE
how to train the network with mini-batches in eager execution?

### ISSUE THREAD

**Issue by alinamadchian** [tensorflow/tensorflow#21663]
### Describe the problem
I find a simple code from tensorflow documents to train a simple neural network by eager execution.
I need to know that how should I modify the code to train the network with mini-batches?
### Source code / logs
from __future__ import absolute_import, division, print_function
import tensorflow as tf
import numpy as np
tfe = tf.contrib.eager
NUM_EXAMPLES = 1000
training_inputs = tf.random_normal([NUM_EXAMPLES])
noise = tf.random_normal([NUM_EXAMPLES])
training_outputs = training_inputs * 3 + 2 

def prediction(input, weight, bias):
  return input * weight + bias

def loss(weights, biases):
  error = prediction(training_inputs, weights, biases) - training_outputs
  return tf.reduce_sum(tf.square(error))

def grad(weights, biases):
  with tf.GradientTape() as tape:
    loss_value = loss(weights, biases)
  return tape.gradient(loss_value, [weights, biases])

train_steps = 20000
learning_rate = 0.01
W = tfe.Variable(5.)
B = tfe.Variable(10.)
optimizer=tf.train.AdamOptimizer(0.1)

for i in range(train_steps):
    grads=grad(W,B)
    optimizer.apply_gradients(zip(grads,[W,B]),global_step=tf.train.get_or_create_global_step())
    if i % 20 == 0:
     print("Loss at step {:03d}: {:.3f}".format(i, loss(W, B)))

**Comment by asimshankar** [comment]
You'd want to structure your code so that each iteration feeds a minibatch instead of hardcoding the input in the call to `loss`. 

See for example the [custom training tutorial](https://colab.research.google.com/github/tensorflow/tensorflow/blob/master/tensorflow/contrib/eager/python/examples/notebooks/custom_training.ipynb) and other tutorial notebooks on www.tensorflow.org/tutorials and the [MNIST example](https://github.com/tensorflow/models/blob/master/official/mnist/mnist_eager.py)

Hope that helps!
